easy-ui-react/src/Notification/Notification.tsx

```diff
@@ -0,0 +1,391 @@
+ *   hasIcon?: boolean;
+ *   // Callback function when dismissing alerts
+ *   onDismiss?: () => void;
+ * }
+ * ```
+ * @example
+ * _Success:_
+```tsx
+* import { useNotification } from "@easypost/easy-ui/Notification";
+* import { Button } from "@easypost/easy-ui/Button";
+*
+* function Component() {
+*  const notification = useNotification();
+*
+*  const onToast = () => {
+*    notification.showSuccessToast({ message: "message" });
+*  };
+*
+*  const onAlert = () => {
+*    notification.showSuccessAlert({ message: "message" });
+* };
+*
+*  return (
+*    <>
+*      <Button onPress={onToast}>Show toast</Button>
+*      <Button onPress={onAlert}>Show alert</Button>
+*    </>
+*  );
+*}
+*```
+* @example
+* _Without status icons:_
+```tsx
+* import { useNotification } from "@easypost/easy-ui/Notification";
+* import { Button } from "@easypost/easy-ui/Button";
+*
+* function Component() {
+*  const notification = useNotification();
+*
+*  const onToast = () => {
+*    notification.showPromotionalToast({ message: "message", hasIcon: false });
+*  };
+*
+*  const onAlert = () => {
+*    notification.showWarningAlert({ message: "message", hasIcon: false });
+* };
+*
+*  return (
+*    <>
+*      <Button onPress={onToast}>Show toast</Button>
+*      <Button onPress={onAlert}>Show alert</Button>
+*    </>
+*  );
+*}
+*```
+* @example
+* _With onDismiss callback, only applicable for alerts:_
+```tsx
+* import { useNotification } from "@easypost/easy-ui/Notification";
+* import { Button } from "@easypost/easy-ui/Button";
+*
+* function Component() {
+*  const notification = useNotification();
+*
+*
+*  const onAlert = () => {
+*    notification.showNeutralAlert({
+*      message: "message",
+*      onDismiss: () => alert("i am called immediately before the alert closes"),
+*    });
+*  };
+*
+*  return (
+*    <>
+*      <Button onPress={onAlert}>Show alert</Button>
+*    </>
+*  );
+*}
+*```
+ */
+export function useNotification() {
+  const notification = useContext(NotificationContext);
+  if (!notification) {
+    throw new Error(
+      "useNotification() must be used within a NotificationProvider, which is bundled with EasyUIProvider",
+    );
+  }
+
+  return notification;
+}
+
+export type NotificationPositionPlacement = {
+  /** Top offset */
+  top?: string;
+  /** Right offset */
+  right?: string;
+  /** Bottom offset */
+  bottom?: string;
+  /** Left offset */
+  left?: string;
+};
+
+export type NotificationPositionType = "fixed" | "absolute";
+
+export type NotificationPlacementProps = {
+  /** HTML ID of element where notifications will render to */
+  htmlId?: string;
+  /** Position type */
+  positionType?: NotificationPositionType;
+  /** Position placement */
+  positionPlacement?: NotificationPositionPlacement;
+};
+
+export type NotificationProviderProps = {
+  /** Child components */
+  children: ReactNode;
+  /** Notification placement props */
+  notificationPlacementProps?: NotificationPlacementProps;
```

Comment:
**suggestion:** nice i like this architecture

just a couple suggestions you can feel free to take or leave:

- allow for accepting an element instead of an id, this way we don't have to presuppose anything about the consumer. they don't have to give the element an id for instance. and it's easy enough to just do `document.getElementById()` on their end or even pass a React `ref.current`
- consider making some names a little more brief

```tsx
<EasyUIProvider
  notificationPlacement={{
    container: document.querySelector('.epr-header'),
    position: "absolute",
    offset: { top: 0, left: 0 }
  }}
>
```

Reply:
- i opted for a `containerFn` prop where the consumer could pass a simple function that gets the element. the idea is when the consumer wants to pass a container, ideally they want the specific styles they pass along to apply as well, but if the container doesn't exist or they pass a bad prop, we have control over the request to default to the original styles if the request fails (of course this could be done another way too). also, we get the added benefit of only issuing the call to get the container when necessary via the `state` prop. let me know if you think theres a downside to this approach though

```tsx
<EasyUIProvider
  notificationPlacement={{
    containerFn: () => document.querySelector('.epr-header'),
    position: "absolute",
    offset: { top: 0, left: 0 }
  }}
>
```

- updated names to be more brief

Reply:
works for me. i think i typically see those functions named as `getContainer: () => document...` but what you have is fine too.

did we want to update to `notificationPlacement` instead of `notificationPlacementProps`? i prefer slightly more brief but up to you